### PR TITLE
Fix trilogy builds

### PR DESCRIPTION
**Pull request by eileencodes**
https://github.com/rails/rails/pull/48007 and https://github.com/rails/rails/pull/47993 combined

### CODE REVIEW

**Comment by byroot** [comment]
Uh?? Why is it still failing? I don't get it

**Comment by casperisfine** [comment]
The paths are lies:

```
Error:
TrilogyAdapterTest#test_#reset_closes_connection_with_existing_connection:
Trilogy::SyscallError::ENOENT: No such file or directory - trilogy_connect - unable to connect to /tmp/mysql.sock
    /usr/local/bundle/bundler/gems/trilogy-6dd14cca18da/contrib/ruby/lib/trilogy.rb:174:in `_initialize'
    /usr/local/bundle/bundler/gems/trilogy-6dd14cca18da/contrib/ruby/lib/trilogy.rb:174:in `initialize'
    /rails/activerecord/test/cases/adapters/trilogy/trilogy_adapter_test.rb:169:in `new'
    /rails/activerecord/test/cases/adapters/trilogy/trilogy_adapter_test.rb:169:in `block in <class:TrilogyAdapterTest>'
 
bin/rails test /rails/activerecord/test/cases/adapters/trilogy/trilogy_adapter_test.rb:168
```

I don't get what's up.

File: activerecord/test/cases/adapters/trilogy/trilogy_adapter_test.rb [hunk 1]
```diff
@@ -9,11 +9,7 @@
 
 class TrilogyAdapterTest < ActiveRecord::TrilogyTestCase
   setup do
-    @configuration = {
-      adapter: "trilogy",
-      username: "rails",
-      database: "activerecord_unittest",
-    }
+    @configuration = ARTest.config.fetch("connections").fetch("trilogy").fetch("arunit").symbolize_keys
```
Comment (byroot) [inline activerecord/test/cases/adapters/trilogy/trilogy_adapter_test.rb:12]: I was missing the `symbolize_keys` 🤦

**Comment by byroot** [comment]
Ok, so it now passed on 3.0 and 3.2, but failed for new reasons on 3.1 and 2.7.

**Comment by casperisfine** [comment]
> failed for new reasons on 3.1 and 2.7.

I can repro locally by using the same seed. So it must be some kid of leak.

**Comment by byroot** [comment]
Ok, so a lot of the tests imported from the gem weren't safe to run as part of the test suite because they mutate the database.

Most were redundant anyway so I just removed all the leaky tests.

**Pull request merged**